src/board.py

```diff
@@ -1,16 +1,49 @@
 """Board Module"""
+from typing import Tuple, List
+from src.coordinate import Coordinate
+from src.snake import Snake
 
 class Board:
     """Track the cooardinates for all snakes and food in the game."""
     def __init__(self, data):
         self._data = data
+        self._snakes = None
+        self._foods = None
 
     @property
-    def snakes(self):
+    def snakes(self) -> List[Snake]:
         """Retreive the list of snakes from the board data."""
-        return self._data['snakes']
+        if self._snakes is None:
+            snakes = []
+            for snake_data in self._data['snakes']:
+                snakes.append(Snake(snake_data))
+            self._snakes = snakes
+        return self._snakes
 
     @property
-    def foods(self):
+    def foods(self) -> List[Coordinate]:
         """Retreive the list of food from the board data."""
-        return self._data['food']
+        if not self._foods or len(self.foods) <= 0:
```

Comment:
the type checks complain about None not matching the datatype of List[type] so I'll make them all consistent - but I think I need to keep the len check, because for example when in snake.py's constructor, it initializes self._coordinates as [], it's not None but it still needs to go into the conditional which creates the _coordinates data